packages/zowe-explorer/src/utils/CertificateWizard.ts

```diff
@@ -0,0 +1,127 @@
+        switch (message.command) {
+            case "promptCert":
+                {
+                    const tempCert = await Gui.showOpenDialog({
+                        title: "Enter the path to the certificate for authenticating the connection.",
+                        canSelectFiles: true,
+                        canSelectFolders: false,
+                        canSelectMany: false,
+                        defaultUri: this.opts.cert ? Uri.file(this.opts.cert) : undefined,
+                        filters: { "Certificate Files": ["cer", "crt", "pem"], "All Files": ["*"] },
+                        openLabel: "Select Certificate",
+                        ...(this.opts.dialogOpts ?? {}),
+                    });
+                    if (tempCert != null && tempCert[0] != null) {
+                        this.opts.cert = tempCert[0].fsPath;
+                        await this.panel.webview.postMessage({
+                            opts: {
+                                cert: tempCert[0].fsPath,
+                            },
+                        });
+                    }
+                }
+                break;
+            case "promptCertKey":
+                {
+                    const tempCertKey = await Gui.showOpenDialog({
+                        title: "Enter the path to the certificate key for authenticating the connection.",
+                        canSelectFiles: true,
+                        canSelectFolders: false,
+                        canSelectMany: false,
+                        defaultUri: this.opts.certKey ? Uri.file(this.opts.certKey) : undefined,
+                        filters: { "Certificate Keys": ["cer", "crt", "pem", "key"], "All Files": ["*"] },
+                        openLabel: "Select Certificate Key",
+                        ...(this.opts.dialogOpts ?? {}),
+                    });
+                    if (tempCertKey != null && tempCertKey[0] != null) {
+                        this.opts.certKey = tempCertKey[0].fsPath;
+                        await this.panel.webview.postMessage({
+                            opts: {
+                                certKey: tempCertKey[0].fsPath,
+                            },
+                        });
+                    }
+                }
+                break;
+            case "submitted":
+                this.userSubmission.resolve({
+                    cert: this.opts.cert,
+                    certKey: this.opts.certKey,
+                });
+                return;
+            case "ready":
+                await this.panel.webview.postMessage({
+                    opts: this.opts,
+                });
+                break;
+            case "close":
+                setImmediate(() => {
+                    this.panel.dispose();
+                });
+                ZoweLogger.trace(localize("zowe.certWizard.dismissed", "User dismissed the certificate wizard."));
```

Comment:
Can't we use the `userDismissed` variable declared above? 😋

Reply:
Yes, thanks for catching this 😅